pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexOpenRedirectRule.java

```diff
@@ -37,15 +39,25 @@ public ApexOpenRedirectRule() {
     }
 
     @Override
-    public Object visit(ASTNewObjectExpression node, Object data) {
-        checkNewObjects(node, data);
-        return data;
-    }
+    public Object visit(ASTUserClass node, Object data) {
+        if (Helper.isTestMethodOrClass(node)) {
+            return data;
+        }
 
-    @Override
-    public Object visit(ASTVariableDeclaration node, Object data) {
-        findSafeLiterals(node);
+        List<ASTVariableDeclaration> variableDecls = node.findDescendantsOfType(ASTVariableDeclaration.class);
+        for (ASTVariableDeclaration varDecl : variableDecls) {
+            findSafeLiterals(varDecl);
+        }
+
+        List<ASTFieldDeclaration> fieldDecl = node.findDescendantsOfType(ASTFieldDeclaration.class);
+        for (ASTFieldDeclaration fDecl : fieldDecl) {
+            findSafeLiterals(fDecl);
+        }
 
+        List<ASTNewObjectExpression> newObjects = node.findDescendantsOfType(ASTNewObjectExpression.class);
+        for (ASTNewObjectExpression newObj : newObjects) {
+            checkNewObjects(newObj, data);
+        }
         return data;
```

Comment:
as with #163 I'd add a `listOfStringLiteralVariables.clear()` here or change `listOfStringLiteralVariables` to a local var.